lib/xwax/lut.h

```diff
@@ -33,10 +42,22 @@ struct lut {
         avail; /* next available slot */
 };
 
+struct lut_mk2 {
+    struct slot_mk2 *slot;
+    slot_no_t *table, /* hash -> slot lookup */
+        avail; /* next available slot */
+};
+
 int lut_init(struct lut *lut, int nslots);
 void lut_clear(struct lut *lut);
 
-void lut_push(struct lut *lut, unsigned int timecode);
-unsigned int lut_lookup(struct lut *lut, unsigned int timecode);
+void lut_push(struct lut *lut, bits_t timecode);
+unsigned int lut_lookup(struct lut *lut, bits_t timecode);
+
+int lut_init_mk2(struct lut_mk2 *lut, int nslots);
+void lut_clear_mk2(struct lut_mk2 *lut);
+
+void lut_push_mk2(struct lut_mk2 *lut, mk2bits_t *timecode);
+slot_no_t lut_lookup_mk2(struct lut_mk2 *lut, mk2bits_t *timecode);
```

Comment:
So what I'm thinking is we separate all of the changes in this file into lut_traktor_mk2.h like you did with the timecoder file.  that will keep this file pristine and unedited from mark's work.  And the corresponding implementation can go into lut_traktor_mk2.c.

Reply:
Put into lut_mk2.c